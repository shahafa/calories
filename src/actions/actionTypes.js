export const RESET_AUTH_STORE = 'RESET_AUTH_STORE';
export const LOGIN_REQUEST = 'LOGIN_REQUEST';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAILURE = 'LOGIN_FAILURE';
export const SIGNUP_REQUEST = 'SIGNUP_REQUEST';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const GET_MEALS_REQUEST = 'GET_MEALS_REQUEST';
export const GET_MEALS_SUCCESS = 'GET_MEALS_SUCCESS';
export const GET_MEALS_FAILURE = 'GET_MEALS_FAILURE';
export const ADD_MEAL_REQUEST = 'ADD_MEAL_REQUEST';
export const ADD_MEAL_SUCCESS = 'ADD_MEAL_SUCCESS';
export const ADD_MEAL_FAILURE = 'ADD_MEAL_FAILURE';
export const DELETE_MEAL_REQUEST = 'DELETE_MEAL_REQUEST';
export const DELETE_MEAL_SUCCESS = 'DELETE_MEAL_SUCCESS';
export const DELETE_MEAL_FAILURE = 'DELETE_MEAL_FAILURE';
export const EDIT_MEAL_REQUEST = 'EDIT_MEAL_REQUEST';
export const EDIT_MEAL_SUCCESS = 'EDIT_MEAL_SUCCESS';
export const EDIT_MEAL_FAILURE = 'EDIT_MEAL_FAILURE';
export const CLOSE_MEALS_ERROR_SNACKBAR = 'CLOSE_MEALS_ERROR_SNACKBAR';
export const SET_FILTER = 'SET_FILTER';

export const GET_USERS_REQUEST = 'GET_USERS_REQUEST';
export const GET_USERS_SUCCESS = 'GET_USERS_SUCCESS';
export const GET_USERS_FAILURE = 'GET_USERS_FAILURE';
export const UPDATE_USERS_ROLE_REQUEST = 'UPDATE_USERS_ROLE_REQUEST';
export const UPDATE_USERS_ROLE_SUCCESS = 'UPDATE_USERS_ROLE_SUCCESS';
export const UPDATE_USERS_ROLE_FAILURE = 'UPDATE_USERS_ROLE_FAILURE';

export const GET_SETTINGS_REQUEST = 'GET_SETTINGS_REQUEST';
export const GET_SETTINGS_SUCCESS = 'GET_SETTINGS_SUCCESS';
export const GET_SETTINGS_FAILURE = 'GET_SETTINGS_FAILURE';
export const SET_SETTINGS_REQUEST = 'SET_SETTINGS_REQUEST';
export const SET_SETTINGS_SUCCESS = 'SET_SETTINGS_SUCCESS';
export const SET_SETTINGS_FAILURE = 'SET_SETTINGS_FAILURE';
export const CLOSE_SETTINGS_ERROR_SNACKBAR = 'CLOSE_SETTINGS_ERROR_SNACKBAR';
