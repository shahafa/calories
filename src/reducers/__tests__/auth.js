import authReducer from '../auth';
import * as types from '../../constants/actionTypes';

describe('auth reducer', () => {
  it('should return the initial state', () => {
    expect(
      authReducer(undefined, {}),
    ).toMatchSnapshot();
  });

  it('should reset the auth state', () => {
    expect(
      authReducer([], {
        type: types.RESET_AUTH_STATE,
      }),
    ).toMatchSnapshot();
  });

  it('should start login request', () => {
    expect(
      authReducer([], {
        type: types.LOGIN_REQUEST,
      }),
    ).toMatchSnapshot();
  });

  it('should set token after successful login', () => {
    expect(
      authReducer([], {
        type: types.LOGIN_SUCCESS,
        payload: {
          code: 1,
          message: 'Login success',
          token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7ImlkIjoiMmM3YTlhOGItNmE5OS00ODM3LWFmMDQtMjMxNDBlMjA1MWUxIiwiZW1haWwiOiJ1c2VyQG1haWwuY29tIiwicm9sZSI6InVzZXIifSwiaWF0IjoxNDk3NDM1NDE2LCJleHAiOjE0OTc1MjE4MTZ9.439LE9pxyTeXrNEqNaWy8wrp3LZHfbnb4EvHIl1gUSM',
        },
      }),
    ).toMatchSnapshot();
  });

  it('should set error after login failure', () => {
    expect(
      authReducer([], {
        type: types.LOGIN_FAILURE,
        payload: {
          response: {
            code: 5,
            message: 'Invalid email or password',
            errors: [],
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should start signup request', () => {
    expect(
      authReducer([], {
        type: types.SIGNUP_REQUEST,
      }),
    ).toMatchSnapshot();
  });

  it('should set token after successful signup', () => {
    expect(
      authReducer([], {
        type: types.SIGNUP_SUCCESS,
        payload: {
          code: 1,
          message: 'Login success',
          token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7ImlkIjoiMmM3YTlhOGItNmE5OS00ODM3LWFmMDQtMjMxNDBlMjA1MWUxIiwiZW1haWwiOiJ1c2VyQG1haWwuY29tIiwicm9sZSI6InVzZXIifSwiaWF0IjoxNDk3NDM1NDE2LCJleHAiOjE0OTc1MjE4MTZ9.439LE9pxyTeXrNEqNaWy8wrp3LZHfbnb4EvHIl1gUSM',
        },
      }),
    ).toMatchSnapshot();
  });

  it('should set error after signup failure', () => {
    expect(
      authReducer([], {
        type: types.SIGNUP_FAILURE,
        payload: {
          response: {
            code: 6,
            message: 'Account with that email address already exists',
            errors: [],
          },
        },
      }),
    ).toMatchSnapshot();
  });
});
