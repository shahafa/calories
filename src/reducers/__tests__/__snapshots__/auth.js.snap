// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducer should reset the auth state 1`] = `
Object {
  "errorCode": null,
  "errorText": "",
  "isAuthenticated": false,
  "isAuthenticating": false,
  "isSigningUp": false,
  "jwtToken": null,
  "user": Object {
    "role": "",
  },
}
`;

exports[`auth reducer should return the initial state 1`] = `
Object {
  "errorCode": null,
  "errorText": "",
  "isAuthenticated": false,
  "isAuthenticating": false,
  "isSigningUp": false,
  "jwtToken": null,
  "user": Object {
    "role": "",
  },
}
`;

exports[`auth reducer should set error after login failure 1`] = `
Object {
  "error": 5,
  "errorText": "Invalid email or password",
  "isAuthenticated": false,
  "isAuthenticating": false,
  "jwtToken": null,
  "user": Object {
    "role": "",
  },
}
`;

exports[`auth reducer should set error after signup failure 1`] = `
Object {
  "error": 6,
  "errorText": "Account with that email address already exists",
  "isAuthenticated": false,
  "isSigningUp": false,
  "jwtToken": null,
  "user": Object {
    "role": "",
  },
}
`;

exports[`auth reducer should set token after successful login 1`] = `
Object {
  "isAuthenticated": true,
  "isAuthenticating": false,
  "jwtToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7ImlkIjoiMmM3YTlhOGItNmE5OS00ODM3LWFmMDQtMjMxNDBlMjA1MWUxIiwiZW1haWwiOiJ1c2VyQG1haWwuY29tIiwicm9sZSI6InVzZXIifSwiaWF0IjoxNDk3NDM1NDE2LCJleHAiOjE0OTc1MjE4MTZ9.439LE9pxyTeXrNEqNaWy8wrp3LZHfbnb4EvHIl1gUSM",
  "user": Object {
    "email": "user@mail.com",
    "id": "2c7a9a8b-6a99-4837-af04-23140e2051e1",
    "role": "user",
  },
}
`;

exports[`auth reducer should set token after successful signup 1`] = `
Object {
  "isAuthenticated": true,
  "isSigningUp": false,
  "jwtToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7ImlkIjoiMmM3YTlhOGItNmE5OS00ODM3LWFmMDQtMjMxNDBlMjA1MWUxIiwiZW1haWwiOiJ1c2VyQG1haWwuY29tIiwicm9sZSI6InVzZXIifSwiaWF0IjoxNDk3NDM1NDE2LCJleHAiOjE0OTc1MjE4MTZ9.439LE9pxyTeXrNEqNaWy8wrp3LZHfbnb4EvHIl1gUSM",
  "user": Object {
    "email": "user@mail.com",
    "id": "2c7a9a8b-6a99-4837-af04-23140e2051e1",
    "role": "user",
  },
}
`;

exports[`auth reducer should start login request 1`] = `
Object {
  "isAuthenticating": true,
}
`;

exports[`auth reducer should start signup request 1`] = `
Object {
  "isSigningUp": true,
}
`;
